packages/lodestar/src/util/validation/attestation.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  EpochContext,
+  getAttestingIndicesFromCommittee,
+  getCurrentSlot,
+  getIndexedAttestation,
+  isValidIndexedAttestation
+} from "@chainsafe/lodestar-beacon-state-transition";
+import {ATTESTATION_PROPAGATION_SLOT_RANGE} from "../../constants";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {Attestation, BeaconState} from "@chainsafe/lodestar-types";
+import {IBeaconDb} from "../../db/api";
+import {assert} from "@chainsafe/lodestar-utils";
+import {processSlots} from "@chainsafe/lodestar-beacon-state-transition/lib/fast/slot";
+
+/**
+ * is ready to be included in block
+ */
+export function hasValidAttestationSlot(config: IBeaconConfig, genesisTime: number, attestation: Attestation): boolean {
+  const currentSlot = getCurrentSlot(config, genesisTime);
+  return attestation.data.slot + ATTESTATION_PROPAGATION_SLOT_RANGE >= currentSlot
+        && currentSlot >= attestation.data.slot;
+}
+
+export function isUnaggregatedAttestation(
+  config: IBeaconConfig, state: BeaconState, epochCtx: EpochContext, attestation: Attestation
+): boolean {
+  if (state.slot < attestation.data.slot) {
+    processSlots(epochCtx, state, attestation.data.slot);
+  }
+  // Make sure this is unaggregated attestation
+  return getAttestingIndicesFromCommittee(
+    epochCtx.getBeaconCommittee(attestation.data.slot, attestation.data.index),
+    attestation.aggregationBits
+  ).length === 1;
+}
+
+export async function isAttestingToValidBlock(db: IBeaconDb, attestation: Attestation): Promise<boolean> {
+  const blockRoot = attestation.data.beaconBlockRoot.valueOf() as Uint8Array;
+  return await db.block.has(blockRoot);
+}
+
+export async function hasValidatorAttestedForThatTargetEpoch(
+  config: IBeaconConfig, db: IBeaconDb, state: BeaconState, epochCtx: EpochContext, attestation: Attestation
+): Promise<boolean> {
+  if (state.slot < attestation.data.slot) {
+    processSlots(epochCtx, state, attestation.data.slot);
+  }
+  const existingAttestations = await db.attestation.geAttestationsByTargetEpoch(
+    attestation.data.target.epoch
+  );
+    // each attestation has only 1 validator index
+  const existingValidatorIndexes = existingAttestations.map(
+    item => getAttestingIndicesFromCommittee(
```

Comment:
This might be more ergonomic as a method on epochCtx. Same with getCommitteeAssignment. (eg: `epochCtx.getAttestingIndices` and `epochCtx.getCommitteeAssignment`)https://github.com/ChainSafe/lodestar/blob/master/packages/lodestar-cli/src/util/canonicalOptions.ts

Maybe we just make an issue for it for now?

Reply:
Yes pls, since there is awfully lot of stubbing required